src/pybamm/expression_tree/averages.py

```diff
@@ -394,13 +406,44 @@ def size_average(
     else:
         if f_a_dist is None:
             geo = pybamm.geometric_parameters
-            R = pybamm.SpatialVariable(
-                "R", domains=symbol.domains, coord_sys="cartesian"
-            )
-            if ["negative particle size"] in symbol.domains.values():
+            if "negative" in symbol.domain[0]:
+                if "primary" in symbol.domain[0]:
+                    R = pybamm.SpatialVariable(
+                        "R_n_prim", domains=symbol.domains, coord_sys="cartesian"
+                    )
+                elif "secondary" in symbol.domain[0]:
+                    R = pybamm.SpatialVariable(
+                        "R_n_sec", domains=symbol.domains, coord_sys="cartesian"
+                    )
+                else:
+                    R = pybamm.SpatialVariable(
+                        "R_n", domains=symbol.domains, coord_sys="cartesian"
+                    )
+            else:
+                if "primary" in symbol.domain[0]:
+                    R = pybamm.SpatialVariable(
+                        "R_p_prim", domains=symbol.domains, coord_sys="cartesian"
+                    )
+                elif "secondary" in symbol.domain[0]:
+                    R = pybamm.SpatialVariable(
+                        "R_p_sec", domains=symbol.domains, coord_sys="cartesian"
+                    )
+                else:
+                    R = pybamm.SpatialVariable(
+                        "R_p", domains=symbol.domains, coord_sys="cartesian"
+                    )
```

Comment:
more of a code style comment but this is more maintainable if you only set the name in the conditional branches and then at the end do `R = SpatialVariable(name, domains=symbol.domains, coord_sys="cartesian")`